common/util.go

```diff
@@ -386,6 +386,40 @@ func WorkflowIDToHistoryShard(
 	return int32(hash%uint32(numberOfShards)) + 1 // ShardID starts with 1
 }
 
+func MapShardID(
```

Comment:
Can you merge with the existing method?
```
func generateShardIDs(localShardId int32, localShardCount int32, remoteShardCount int32) []int32 {
	var pollingShards []int32
	if remoteShardCount <= localShardCount {
		if localShardId <= remoteShardCount {
			pollingShards = append(pollingShards, localShardId)
		}
		return pollingShards
	}
	// remoteShardCount > localShardCount, replication poller will poll from multiple remote shard.
	for i := localShardId; i <= remoteShardCount; i += localShardCount {
		pollingShards = append(pollingShards, i)
	}
	return pollingShards
}
```

Reply:
the behavior is actually different, generateShardIDs will not create a replication channel in this case
e.g. 

64K shards pulll 16K shards, caller being 16K+1